pkg/sql/session.go

```diff
@@ -1383,15 +1385,41 @@ func (scc *schemaChangerCollection) execSchemaChanges(
 
 // maybeRecover catches SQL panics and does some log reporting before
 // propagating the panic further.
+//
 // TODO(knz): this is where we can place code to recover from
 // recoverable panics.
 func (s *Session) maybeRecover(action, stmt string) {
 	if r := recover(); r != nil {
-		err := errors.Errorf("panic while %s %q: %s", action, stmt, r)
+		var anonymized []string
+		{
+			stmts, err := parser.Parse(stmt)
+			if err == nil {
+				var buf bytes.Buffer
+				for _, stmt := range NewStatementList(stmts) {
+					buf.Reset()
+					tree.FormatNode(&buf, tree.FmtHideConstants, stmt.AST)
```

Comment:
I think this might need to be `FmtAnonymize` as well, which hides table and database names.